lib/improver/utilities/cube_metadata.py

```diff
@@ -644,3 +644,32 @@ def in_vicinity_name_format(cube_name):
                          cube_name[relative_to_threshold_index:])
 
     return new_cube_name
+
+
+def extract_diagnostic_name(cube_name):
```

Comment:
Does there need to be a unit test to check for this? I can't see one that has been written to check extract_diagnostic_name?

Reply:
There is a test at the bottom of test_cube_metadata - `Test_extract_diagnostic_name`.  Is that what you mean?

Reply:
Yes you are right, I think I just couldn't see it because I'd been looking at the files for so long. This all looks fine to me. Thanks!